packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -1084,3 +1085,91 @@ describeCompat("IdCompressor Summaries", "NoCompat", (getTestObjectProvider, com
 		assert(dds2.id.length > 8, "DDS's ID in attached container should not be short");
 	});
 });
+
+const dataStoreCount = 13;
+describeCompat("IdCompressor basic", "NoCompat", (getTestObjectProvider, apis) => {
+	const {
+		dataRuntime: { DataObject, DataObjectFactory },
+		containerRuntime: { ContainerRuntimeFactoryWithDefaultDataStore },
+	} = apis;
+	class TestDataObject extends DataObject {
+		public get _root() {
+			return this.root;
+		}
+
+		public get _context() {
+			return this.context;
+		}
+	}
+
+	const configProvider = createTestConfigProvider();
+	let provider: ITestObjectProvider;
+	const defaultFactory = new DataObjectFactory("T", TestDataObject, [], []);
+
+	const runtimeOptions: IContainerRuntimeOptions = {
+		enableRuntimeIdCompressor: "on",
+	};
+
+	const runtimeFactory = createContainerRuntimeFactoryWithDefaultDataStore(
+		ContainerRuntimeFactoryWithDefaultDataStore,
+		{
+			defaultFactory,
+			registryEntries: [[defaultFactory.type, Promise.resolve(defaultFactory)]],
+			runtimeOptions,
+		},
+	);
+
+	const createContainer = async (): Promise<IContainer> =>
+		provider.createDetachedContainer(runtimeFactory, { configProvider });
+
+	beforeEach("getTestObjectProvider", async function () {
+		provider = getTestObjectProvider();
+		// ODSP specific bug
+		if (provider.driver.type !== "odsp") {
+			this.skip();
+		}
+		configProvider.set("Fluid.ContainerRuntime.IdCompressorEnabled", true);
+		configProvider.set("Fluid.Runtime.UseShortIds", true);
+	});
+
+	function assertInvert(value: boolean, message: string) {
+		assert(!value, message);
+	}
+
+	itExpects(
+		"Id compressor bug repo when id is `[` which encodes to `%5B`",
+		[{ eventName: "fluid:telemetry:Container:ContainerClose", error: "No context for op" }],
+		async () => {
+			const container = await createContainer();
+			const dataObject = (await container.getEntryPoint()) as TestDataObject;
+			const containerRuntime = dataObject._context.containerRuntime;
+			for (let i = 0; i < dataStoreCount; i++) {
+				const createdObject = await defaultFactory.createInstance(containerRuntime);
+				dataObject._root.set(`${i}`, createdObject.handle);
```

Comment:
Suggestion:
Use the following so that you can retrieve data object with id `[` instead of 12 which seems unnatural:

`dataObject._root.set(createdObject.context.id, createdObject.handle);`

Reply:
Then when retrieving it, you could do `dataObject.root.get("[")`

Reply:
Updated